rpm/Makefile

```diff
@@ -11,17 +11,23 @@ source:
 ifneq ($(filter epel-6 amzn-%,$(DISTRO)),)
```

Comment:
The nested if/else conditions in line 11 - 17 look a bit confusing to me. Am I understanding the syntax wrong? Maybe there is some negation involved that I'm not aware?

```
// If it's not equal to epel-6 or amzn-%?
ifneq ($(filter epel-6 amzn-%,$(DISTRO)),)
  CURL_VERSION=7.34.0
else // if it's sles, we should never reach this "else" clause, right?
  ifeq ($(filter sles%,$(DISTRO)),) // Then how can it be equal to sles and enter this "if" clause?
    CURL_VERSION=7.52.1
  endif
endif
```

Reply:
Your thinking is inverted. [`$(filter)`](https://www.gnu.org/software/make/manual/html_node/Text-Functions.html) will return all matching elements from the list, or "" if empty. So this `ifneq (<result_of_filter>,<empty_string>)` checks that the result of `$(filter)` is non-empty (i.e., that there's a match). The next `ifeq` checks that the result is empty (i.e., no match). I've added comments to that effect.

Reply:
Oh geez, I see. Thanks for the clarification.